azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -319,9 +320,15 @@ def __init__(self, xml_text):
         plugin_settings = findall(plugin_settings_list, "Plugin")
 
         for plugin in plugins:
-            ext_handler = ExtensionsConfig._parse_plugin(plugin)
+
+            ext_handler = ExtHandler()
```

Comment:
Let's consider failing the entire goal state if there is an error in any of the extensions.

My concern is that by trying to recover from errors in CRP, skip the incorrect extensions and execute the rest,  we may end up doing the wrong thing.

Reply:
@narrieta could you expand on what you mean by this? Do you mean that the user might have extensions that make sense together, but not necessarily without all of them? Or is it more a concern about the goal state potentially being read incorrectly if some of the extensions are incorrect?

Reply:
Synced offline about it, and we decided to **only** fail the extension with the bad config rather than the whole GoalState. 

 >Do you mean that the user might have extensions that make sense together, but not necessarily without all of them?

The concern was about extensions that have external dependencies where they specifically set their ARM template having our sequential execution in mind. I can sync you up more about this on a call if you want :)